types/foundry/client/canvas/layers/tokens.d.mts

```diff
@@ -0,0 +1,187 @@
+import { TokenPlannedMovement } from "@client/_module.mjs";
+import TokenHUD from "@client/applications/hud/token-hud.mjs";
+import { SceneControl } from "@client/applications/ui/scene-controls.mjs";
+import User from "@client/documents/user.mjs";
+import { ElevatedPoint, Point, Rectangle } from "@common/_types.mjs";
+import { CanvasQuadtree } from "../geometry/_module.mjs";
+import { Token } from "../placeables/_module.mjs";
+import { CanvasHistoryEvent, PlaceablesLayerOptions } from "./_types.mjs";
+import PlaceablesLayer, { MinusOneToOne, PlaceablesLayerPointerEvent } from "./base/placeables-layer.mjs";
+
+export type TokenOcclusionMode = (typeof CONST.TOKEN_OCCLUSION_MODES)[keyof typeof CONST.TOKEN_OCCLUSION_MODES];
+
+/**
+ * The Tokens Container.
+ * @category Canvas
+ */
+export default class TokenLayer<TObject extends Token = Token> extends PlaceablesLayer<TObject> {
+    override quadtree: CanvasQuadtree<TObject>;
+
+    /**
+     * The ruler paths.
+     * @internal
+     */
+    _rulerPaths: PIXI.Container;
+
+    /**
+     * The current index position in the tab cycle
+     * @internal
+     */
+    _tabIndex: number | null;
+
+    /**
+     * The Token that the drag workflow was initiated on, if there's a drag workflow in progress.
+     * Set in {@link foundry.canvas.placeables.Token#_onDragLeftStart} and
+     * {@link foundry.canvas.placeables.Token#_onDragLeftCancel}.
+     * @internal
+     */
+    _draggedToken: TObject | null;
+
+    /**
+     * The currently selected movement action override.
+     * @internal
+     */
+    _dragMovementAction: string | null;
+
+    static override get layerOptions(): PlaceablesLayerOptions;
+
+    static override documentName: "Token";
+
+    /** The set of tokens that trigger occlusion (a union of {@link CONST.TOKEN_OCCLUSION_MODES}). */
+    get occlusionMode(): TokenOcclusionMode;
+
+    set occlusionMode(value: TokenOcclusionMode);
+
+    /** @inheritdoc */
+    override get hookName(): string;
+
+    /* -------------------------------------------- */
+    /*  Properties
+    /* -------------------------------------------- */
+
+    override get hud(): TokenHUD;
+
+    /** An Array of tokens which belong to actors which are owned  */
+    get ownedTokens(): TObject[];
+
+    /** A Set of Token objects which currently display a combat turn marker. */
+    turnMarkers: Set<TObject>;
+
+    /* -------------------------------------------- */
+    /*  Methods
+    /* -------------------------------------------- */
+
+    override getSnappedPoint(point: Point): Point;
+
+    override _prepareKeyboardMovementUpdates(
+        objects: TObject[],
+        dx: MinusOneToOne,
+        dy: MinusOneToOne,
+        dz: MinusOneToOne,
+    ): ({ _id: string } & ElevatedPoint)[];
+
+    override _draw(options: object): Promise<void>;
```

Comment:
```suggestion
    protected override _draw(options: object): Promise<void>;
```

Check rest

Reply:
I found some more. Hopefully I got them all this time.